src/hotspot/share/runtime/javaThread.hpp

```diff
@@ -642,8 +643,11 @@ class JavaThread: public Thread {
     return _carrier_thread_suspended;
   }
 
-  bool is_in_VTMS_transition() const             { return _is_in_VTMS_transition; }
+  bool is_in_non_tmp_VTMS_transition() const     { return _is_in_VTMS_transition; }
+  bool is_in_VTMS_transition() const             { return _is_in_VTMS_transition || _is_in_tmp_VTMS_transition; }
   void set_is_in_VTMS_transition(bool val);
+  bool is_in_tmp_VTMS_transition() const         { return _is_in_tmp_VTMS_transition; }
+  void toggle_is_in_tmp_VTMS_transition()        { _is_in_tmp_VTMS_transition = !_is_in_tmp_VTMS_transition; };
```

Comment:
The naming of the flags does not match up with the naming of the APIs, which is confusing.  An API named  is_in_VTMS_transition() should return _is_in_VTMS_transition. I think part of problem is that _is_in_VTMS_transition is not a superset of _is_in_tmp_VTMS_transition. The flags are never both set true, although both can be false. Maybe this should be changed to make it an invariant that if _is_in_tmp_VTMS_transit is true, then _is_in_tmp_VTMS_transition is true.

Reply:
I agree, this naming is not good. However, I was reluctant to do required renaming because it can potentially impact many spots and make review harder. Let me think a little bit more on this.

Reply:
I've simplified this naming but, probably, not in a way your were expecting. Please, see my comment below.